script.js

```diff
@@ -1,12 +1,138 @@
+      "0" + episode.number
+    ).slice(-2)}`;
+
+    episodeNumber.innerHTML = `${episodeCode}`;
+    episodeTitle.innerHTML = `${episode.name}`;
+    episodeImage.src = episode.image.medium;
+    episodeSummery.innerHTML = episode.summary;
+
+    episodeCard.appendChild(episodeNumber);
+    episodeCard.appendChild(episodeTitle);
+    episodeCard.appendChild(episodeImage);
+    episodeCard.appendChild(episodeSummery);
+    list.appendChild(episodeCard);
+    count++;
+  });
 }
 
-window.onload = setup;
+// Display episodes with search box function:
+
+function displayEpisodesWithSearchBox(episodes) {
+  //Callback functions
+  displayDropBox(episodes);
+  displayEpisodes(episodes);
+
+  countSpan.innerHTML = `Displaying ${count} / ${episodes.length} episodes`;
+
+  searchInput.addEventListener("input", (e) => {
+    const inputValue = e.target.value.toLowerCase();
+    console.log("inputValue");
+    list.innerHTML = "";
+    let count = 0;
+
+    episodes.forEach((episode) => {
+      if (
+        episode.name.toLowerCase().includes(inputValue) ||
+        episode.summary.toLowerCase().includes(inputValue)
+      ) {
+        let episodeCard = document.createElement("li");
+        let episodeNumber = document.createElement("h3");
+        let episodeTitle = document.createElement("h3");
+        let episodeImage = document.createElement("img");
+        let episodeSummery = document.createElement("p");
+        let episodeCode = `S ${("0" + episode.season).slice(-2)} - E ${(
+          "0" + episode.number
+        ).slice(-2)}`;
+
+        episodeNumber.innerHTML = `${episodeCode}`;
+        episodeTitle.innerHTML = `${episode.name}`;
+        episodeImage.src = episode.image.medium;
+        episodeSummery.innerHTML = episode.summary;
+
+        episodeCard.appendChild(episodeNumber);
+        episodeCard.appendChild(episodeTitle);
+        episodeCard.appendChild(episodeImage);
+        episodeCard.appendChild(episodeSummery);
+        list.appendChild(episodeCard);
+
+        count++;
+      }
+    });
+    countSpan.innerHTML = `Displaying ${count} / ${episodes.length} episodes`;
+  });
+}
+
+// Dropbox and search bar:
+
+function displayDropBox(episodes) {
+  allOptions.value = "All";
+  allOptions.innerHTML = "All episodes";
+  select.appendChild(allOptions);
+
+  episodes.forEach((episode) => {
+    let episodeCode = `S ${("0" + episode.season).slice(-2)} E ${(
+      "0" + episode.number
+    ).slice(-2)}`;
+
+    const eachOption = document.createElement("option");
+    eachOption.value = episode.id;
+    eachOption.innerHTML = `${episodeCode} - ${episode.name}`;
+    select.appendChild(eachOption);
+  });
+
+  select.addEventListener("change", () => {
+    let newArray = [];
+    if (select.value === "All") {
+      newArray = episodes;
+    } else {
+      newArray = episodes.filter((episode) =>
+        select.value.includes(episode.id)
+      );
+    }
+    let count = 0;
```

Comment:
instead of using a counter here you can use the newArray length. (which anyway should always be one)